src/app-layer-htp-file.c

```diff
@@ -292,6 +360,12 @@ int HTPFileStoreChunk(HtpState *s, const uint8_t *data, uint32_t data_len,
         goto end;
     }
 
+    if (s->file_range_ids != 0) {
+        result = FileAppendDataById(files, s->file_range_ids, data, data_len);
```

Comment:
I think it would be useful if the commit message explained the overall logic of the modifications, as I'm a bit confused with why some things are apparently being done twice. Like 2 calls to `FileAppendDataById` in a row here.

Reply:
I am adding the comments of the PR

Reply:
Also explain the overall design in the commit message too please.